src/server/ns_turn_server.c

```diff
@@ -3905,6 +3908,15 @@ static int handle_turn_command(turn_turnserver *server, ts_ur_super_session *ss,
 
     *resp_constructed = 1;
   }
+  if(err_code == 401 && *server->ratelimit_401_requests_per_window > 0) {
+      ioa_addr *rate_limit_address = get_remote_addr_from_ioa_socket(ss->client_socket);
+      if (ratelimit_is_address_limited(rate_limit_address, *server->ratelimit_401_requests_per_window, *server->ratelimit_401_window_seconds)) {
+          no_response = 1;
+          char raddr[129];
+          addr_to_string_no_port(rate_limit_address, (unsigned char *)raddr);
+          TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "401 rate limit exceeded from %s, response not sent\n", raddr);
```

Comment:
Would it make sense to add the remote address to the "Cannot find credentials" log as well, which is shown before the rate limit kicks in?

https://github.com/coturn/coturn/blob/f6004a1c185d666ebef024e54501266bfbd333ed/src/server/ns_turn_server.c#L3475